src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php

```diff
@@ -46,7 +46,7 @@ abstract class AbstractFailedMessagesCommand extends Command
     /**
      * @param ServiceProviderInterface $failureTransports
      */
-    public function __construct(?string $globalFailureReceiverName, $failureTransports)
+    public function __construct(?string $globalFailureReceiverName, ?ServiceProviderInterface $failureTransports)
     {
         $this->failureTransports = $failureTransports;
         if (!$failureTransports instanceof ServiceProviderInterface) {
```

Comment:
Can't we remove this BC layer? Same for the `getReceiverName()` method below?

Reply:
But you are not always required to pass a `ServiceProviderInterface`, right? 

Are you still allowed to do null? 

I started working on removing this, but I didn't quite understand what the end result should look like. I either need some guidance or I'll leave this to another PR (which Im currently doing)

Reply:
This is handled in #41878